From 99e5e148ae155e8ddb769a3e8734583d5b0a8afb Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Fri, 20 Nov 2015 10:43:04 +0800
Subject: [PATCH 19/28] imx227: add 1080P 30FPS setting

Add imx227 1080p 30fps for 1080P 30fps video recording

Change-Id: I2da64aa2a1471e0481e7d917a5b755158b07c54e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: #H1504085120
Signed-off-by: Bin Han <bin.b.han@intel.com>
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/85858
Build: Dessert M COHO (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/442542
---
 .../external_drivers/camera/drivers/media/i2c/imx/imx227.h  | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/imx/imx227.h b/drivers/external_drivers/camera/drivers/media/i2c/imx/imx227.h
index ccb6a7a..10e5b86 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/imx/imx227.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/imx/imx227.h
@@ -101,7 +101,7 @@ static struct imx_reg const imx_VIDEO_2M_9X16_45fps[] = {
 	{IMX_8BIT, 0x0112, 0x0a},
 	{IMX_8BIT, 0x0113, 0x0a},
 	{IMX_8BIT, 0x0344, 0x02},
-	{IMX_8BIT, 0x0345, 0x8e},
+	{IMX_8BIT, 0x0345, 0x8a},
 	{IMX_8BIT, 0x0346, 0x01},
 	{IMX_8BIT, 0x0347, 0x88},
 	{IMX_8BIT, 0x0348, 0x06},
@@ -109,7 +109,7 @@ static struct imx_reg const imx_VIDEO_2M_9X16_45fps[] = {
 	{IMX_8BIT, 0x034a, 0x09},
 	{IMX_8BIT, 0x034b, 0x17},
 	{IMX_8BIT, 0x034c, 0x04},
-	{IMX_8BIT, 0x034d, 0x44},
+	{IMX_8BIT, 0x034d, 0x48},
 	{IMX_8BIT, 0x034e, 0x07},
 	{IMX_8BIT, 0x034f, 0x90},
 
@@ -127,7 +127,7 @@ static struct imx_reg const imx_VIDEO_2M_9X16_45fps[] = {
 	{IMX_8BIT, 0x040a, 0x00},
 	{IMX_8BIT, 0x040b, 0x00},
 	{IMX_8BIT, 0x040c, 0x04},
-	{IMX_8BIT, 0x040d, 0x44},
+	{IMX_8BIT, 0x040d, 0x48},
 	{IMX_8BIT, 0x040e, 0x07},
 	{IMX_8BIT, 0x040f, 0x90},
 
@@ -594,7 +594,7 @@ struct imx_resolution imx227_res_video[] = {
 	{
 		.desc = "imx_VIDEO_2M_9X16_45fps",
 		.regs = imx_VIDEO_2M_9X16_45fps,
-		.width = 1092,
+		.width = 1096,
 		.height = 1936,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
@@ -602,6 +602,11 @@ struct imx_resolution imx227_res_video[] = {
 		.used = 0,
 		.fps_options = {
 			{
+				.fps = 30,
+				.pixels_per_line = 0x0E70,
+				.lines_per_frame = 0x0C08,
+			},
+			{
 				.fps = 45,
 				.pixels_per_line = 0x0E70,
 				.lines_per_frame = 0x0800,
-- 
1.9.1

